database/database.go

```diff
@@ -192,6 +192,38 @@ func (db *DB) CreateBlockedSkylink(ctx context.Context, skylink *BlockedSkylink)
 	return nil
 }
 
+// CreateBlockedSkylinkBulk creates new blocked skylinks in bulk. It returns the
+// number of created entries.
+func (db *DB) CreateBlockedSkylinkBulk(ctx context.Context, skylinks []BlockedSkylink) (int, error) {
+	// Convenience variables
+	logger := db.staticLogger
+
+	// Ensure all required properties are set on the given blocked skylinks
+	for _, skylink := range skylinks {
+		err := skylink.Validate()
+		if err != nil {
+			return 0, errors.AddContext(err, "unexpected blocked skylink")
+		}
+	}
+
+	// Convert the given array to an interface array
+	docs := make([]interface{}, len(skylinks))
+	for i, doc := range skylinks {
+		docs[i] = doc
+	}
+
+	// Insert all objects in the database
+	opts := options.InsertMany()
+	opts.SetOrdered(false)
+	res, err := db.staticSkylinks.InsertMany(ctx, docs, opts)
+	if err != nil && !isDuplicateKey(err) {
```

Comment:
Just to verify. When I write 3 entries. One of them is a duplicate. Another one failed because of a different reason. 
Then this will not print the error for the entry that failed without the duplicate error right? We also won't return an error here. So as long as there is a single duplicate key error here it overshadows any other error that might happen.

Reply:
Ok, point taken. Not an issue! I can simply cast it to the interface and range over all write errors